File: lib/models/fbs.dart
class FBS {
  final int id;
  final int userId;
  final DateTime testDate;
  final double fbsLevel;
  final String? imageUrl;

  FBS({
    required this.id,
    required this.userId,
    required this.testDate,
    required this.fbsLevel,
    this.imageUrl,
  });

  // Factory constructor to create an FBS instance from a map (JSON)
  factory FBS.fromJson(Map<String, dynamic> json) {
    return FBS(
      id: json['id'] as int,
      userId: json['user_id'] as int,
      testDate: DateTime.parse(json['test_date'] as String),
      fbsLevel: json['fbs_level'] as double,
      imageUrl: json['image_url'] as String?,
    );
  }

  // Method to convert an FBS instance to a map (JSON)
  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'user_id': userId,
      'test_date': testDate.toIso8601String(),
      'fbs_level': fbsLevel,
      'image_url': imageUrl,
    };
  }
}

File: models-kotlin/FBS.kt
package com.healthtracker.models

import com.google.gson.annotations.SerializedName
import java.time.LocalDateTime
import java.time.format.DateTimeFormatter

data class FBS(
    val id: Int,
    @SerializedName("user_id")
    val userId: Int,
    @SerializedName("test_date")
    val testDate: LocalDateTime,
    @SerializedName("fbs_level")
    val fbsLevel: Double,
    @SerializedName("image_url")
    val imageUrl: String? = null
) {
    companion object {
        private val dateFormatter = DateTimeFormatter.ISO_LOCAL_DATE_TIME

        fun fromJson(json: Map<String, Any?>): FBS {
            return FBS(
                id = json["id"] as Int,
                userId = json["user_id"] as Int,
                testDate = LocalDateTime.parse(json["test_date"] as String, dateFormatter),
                fbsLevel = json["fbs_level"] as Double,
                imageUrl = json["image_url"] as String?
            )
        }
    }

    fun toJson(): Map<String, Any?> {
        return mapOf(
            "id" to id,
            "user_id" to userId,
            "test_date" to testDate.format(dateFormatter),
            "fbs_level" to fbsLevel,
            "image_url" to imageUrl
        )
    }
}